### PR TITLE
Add option to include the exported types

### LINKED ISSUE #94: Option to include exported types
To get closer to the output of `go doc`, include the types of the package
i.e.
`type Config struct{ ... }`

### PR DESCRIPTION
**Pull request by romainbou**
New feature to include the types #94 using the `-types` argument
It replaces the body of structs by an ellipsis "...", similarly to `go doc`

New types section like this:

### type [Config](/goreadme.go#L126)

`type Config struct { ... }`

### type [GoReadme](/goreadme.go#L121)

`type GoReadme struct { ... }`

GoReadme enables getting readme.md text from a go package.

Fixes #94

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -16,6 +16,7 @@ github.com/go-stack/stack v1.6.0/go.mod h1:v0f6uXyyMGvRgIKkXu+yp6POWl0qKG85gN/me
 github.com/goccy/go-yaml v1.4.3/go.mod h1:PsEEJ29nIFZL07P/c8dv4P6rQkVFFXafQee85U+ERHA=
 github.com/golang/gddo v0.0.0-20200324184333-3c2cc9a6329d h1:ZJhGJay808i+klrJbox3i5NMVerJ3/tEhtOTeQpPwJQ=
 github.com/golang/gddo v0.0.0-20200324184333-3c2cc9a6329d/go.mod h1:sam69Hju0uq+5uvLJUMDlsKlQ21Vrs1Kd/1YFPNYdOU=
+github.com/golang/gddo v0.0.0-20200831202555-721e228c7686 h1:5vu7C+63KTbsSNnLhrgB98Sqy8MNVSW8FdhkcWA/3Rk=
```
Comment (posener) [inline go.sum:19]: Can you please run `go mod tidy`?

File: goreadme.go [hunk 2]
```diff
@@ -207,6 +209,19 @@ func (r *GoReadme) get(ctx context.Context, name string) (*pkg, error) {
 			}
 		}
 	}
+	if !r.config.Types {
```
Comment (posener) [inline goreadme.go:212]: Can you add a similar comment to the one above?

```go
	// If types were not requested to be added to the readme, add their
	// examples to the main readme.
```

**Review by posener — COMMENTED** [review]
Thanks!

Can you add tests? For example, add a type definition [here](https://github.com/posener/goreadme/blob/master/testdata/pkg1/pkg1.go)?

File: go.sum (hunk 1 above)
Comment (romainbou) [reply go.sum]: Sure thing!

File: goreadme.go (hunk 2 above)
Comment (romainbou) [reply goreadme.go:215]: Yes

**Comment by romainbou** [comment]
Yes indeed, I added a type to [pkg1](https://github.com/posener/goreadme/blob/master/testdata/pkg1/pkg1.go) and a new [pkg8_types](https://github.com/posener/goreadme/blob/master/testdata/pkg8_types/pkg.go) just for the types

**Review by posener — APPROVED** [review]

**Comment by posener** [comment]
Fixes #94

**Pull request merged**